parlai/tasks/genderation_bias/agents.py

```diff
@@ -0,0 +1,327 @@
+            type=int,
+            default=-1,
+            help='Maximum number of utterances to include per episode. '
+            'Default -1 keeps all.',
+        )
+        agent = parser.add_argument_group('Controllable Gen Teacher Args')
+        agent.add_argument(
+            '--invalidate-cache',
+            type='bool',
+            default=False,
+            help='Set this to True to rebuild the data (may want to do this if '
+            'original data has changed or you want to rebuild with new options)',
+        )
+        agent.add_argument(
+            '--max-examples',
+            type=int,
+            default=-1,
+            help='If greater than zero, will stop building after a certain num of exs',
+        )
+        agent.add_argument(
+            '--four-class',
+            type='bool',
+            default=True,
+            help='Use 4 class classification for gendered word list (f0m0/f0m1/f1m0/f1m1). '
+            'Set false to use 3-class classification (MALE/FEMALE/NEUTRAL)',
+        )
+        agent.add_argument(
+            '--fixed-control',
+            type=str,
+            default='',
+            help='Always append this fixed control variable, good for deploy time.',
+        )
+        # Add the arguments for the task teacher
+        opt = parser.parse_and_process_known_args()[0]
+        tasks = get_original_task_module(opt, multi_possible=True)
+        for task in tasks:
+            if hasattr(task, 'add_cmdline_args'):
+                task.add_cmdline_args(parser)
+
+        return parser
+
+    def __init__(self, opt: Opt, shared: TShared = None):
+        assert opt['flatten_delimiter'] == opt.get(
+            'delimiter', '\n'
+        ), '--flatten-delimiter and --delimiter are set differently, please inspect and set to the same to avoid unexpected results'
+        self.four_class = opt['four_class']
+        self.opt = opt
+
+        if shared and 'data' in shared:
+            self.data = shared['data']
+        else:
+            self.word_lists = self.build_wordlists(opt)
+            self.data = self._setup_data(opt)
+
+        super().__init__(opt, shared)
+        self.reset()
+
+    def num_episodes(self) -> int:
+        return len(self.data)
+
+    def num_examples(self) -> int:
+        return len(self.data)
+
+    def _get_save_path(self, datapath: str, date: str) -> str:
+        """
+        Return save path for the controllable gen data.
+
+        :param datapath:
+            path to ParlAI Data
+        :param date:
+            current date
+
+        :return path:
+            return path to save
+        """
+        return os.path.join(
+            datapath,
+            f"{self.original_task_name.replace(':', '_')}_flattened_controllable_gen_{date}",
+        )
+
+    @classmethod
+    def build_wordlists(cls, opt: Opt) -> Tuple[List[str], List[str]]:
+        """
+        Load list of explicitly gendered words from.
```

Comment:
Hmm it looks like this sentence got split in two

Reply:
oh this might be black adding a period to the end of the top line of the docstring